ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/CommandService.scala

```diff
@@ -0,0 +1,221 @@
+    for {
+      alice <- allocateParty()
+      transaction <- session.bindings.submitAndWaitForTransaction(
+        alice,
+        context.applicationId,
+        context.nextCommandId(),
+        Seq(Dummy(alice).create.command.command))
+    } yield {
+      assert(
+        transaction.transactionId.nonEmpty,
+        s"The returned transaction ID should be a non empty string, but was '${transaction.transactionId}'.")
+
+      assert(
+        transaction.events.size == 1,
+        s"The returned transaction should contain 1 event, but contained ${transaction.events.size}")
+
+      val event = transaction.events.head
+      assert(
+        event.event.isCreated,
+        s"The returned transaction should contain a created-event, but was ${event.event}"
+      )
+
+      assert(
+        event.getCreated.getTemplateId == Dummy.id.unwrap,
+        s"The template ID of the created-event should by ${Dummy.id}, but was ${event.getCreated.getTemplateId}"
+      )
+    }
+  }
+
+  private val submitAndWaitForTransactionTree = LedgerTest(
+    "CSsubmitAndWaitForTransactionTree",
+    "SubmitAndWaitForTransactionTree returns a transaction tree") { implicit context =>
+    for {
+      alice <- allocateParty()
+      transactionTree <- session.bindings.submitAndWaitForTransactionTree(
+        alice,
+        context.applicationId,
+        context.nextCommandId(),
+        Seq(Dummy(alice).create.command.command))
+    } yield {
+      assert(
+        transactionTree.transactionId.nonEmpty,
+        s"The returned transaction ID should be a non empty string, but was '${transactionTree.transactionId}'.")
+
+      assert(
+        transactionTree.eventsById.size == 1,
+        s"The returned transaction tree should contain 1 event, but contained ${transactionTree.eventsById.size}")
+
+      val event = transactionTree.eventsById.head._2
+      assert(
+        event.kind.isCreated,
+        s"The returned transaction tree should contain a created-event, but was ${event.kind}")
+
+      assert(
+        event.getCreated.getTemplateId == Dummy.id.unwrap,
+        s"The template ID of the created-event should by ${Dummy.id}, but was ${event.getCreated.getTemplateId}"
+      )
+    }
+  }
+
+  private val resendingSubmitAndWait = LedgerTest(
+    "CSduplicateSubmitAndWait",
+    "SubmitAndWait returns Empty for a duplicate submission") { implicit context =>
+    for {
+      alice <- allocateParty()
+      commandId = context.nextCommandId()
+      empty1 <- session.bindings.submitAndWait(
+        alice,
+        context.applicationId,
+        commandId,
+        Seq(Dummy(alice).create.command.command))
+      empty2 <- session.bindings.submitAndWait(
+        alice,
+        context.applicationId,
+        commandId,
+        Seq(Dummy(alice).create.command.command))
+      transactions <- flatTransactions(alice)
+    } yield {
+      assert(empty1 == empty2, "The responses of SubmitAndWait did not match.")
```

Comment:
I'm not entirely sure of the usefulness of this check. It also doesn't seem to match the test description.